util/util.c

```diff
@@ -83,6 +83,35 @@ uint32_t find_last_set_bit(uint32_t word)
 	return last_set_bit;
 }
 
+/**
+ * Locate the closest element in an array.
+ */
+uint32_t find_closest(int32_t val,
+		      const int32_t *array,
+		      uint32_t size)
+{
+	uint32_t diff = abs((abs(array[0]) - abs(val)));
+	uint32_t ret = 0;
+	uint32_t i;
+
+	for (i = 1; i < size; i++) {
+		if (abs((abs(array[i]) - abs(val))) < diff) {
+			diff = abs((abs(array[i]) - abs(val)));
+			ret = i;
+		}
+	}
+
+	return ret;
+}
+
+/**
+ * Shift the value and apply the specified mask.
+ */
+uint32_t field_prep(uint32_t mask, uint32_t val)
```

Comment:
For performance, I think it will be better if this function will be a define or at least inline if type check is needed

Reply:
@dbogdan Do we really need it as a function or as a macro ? I can see the register bit definitions using an equivalent macro FIELD_PREP like on linux, but what is the usecase where we'd need a function ?

Reply:
For consistency with the rest of the file, I would keep it like this. Another commit/pull-request can implement these improvements for all the functionts.